tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,207 @@ def __init__(self, **kwargs):
 
         kwargs["groups"] = -1
         super().__init__(**kwargs)
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class FilterResponseNormalization(tf.keras.layers.Layer):
+    """Filter response normalization layer.
+
+    Filter Response Normalization (FRN), a normalization
+    method that enables models trained with per-channel
+    normalization to achieve high accuracy. It performs better than
+    all other normalization techniques for small batches and is par
+    with Batch Normalization for bigger batch sizes.
+
+    Arguments
+        axis: List of axes that should be normalized. This should represent the
+        spatial dimensions.
+        epsilon: Small float value added to variance to avoid dividing by zero.
+        beta_initializer: Initializer for the beta weight.
+        gamma_initializer: Initializer for the gamma weight.
+        beta_regularizer: Optional regularizer for the beta weight.
+        gamma_regularizer: Optional regularizer for the gamma weight.
+        beta_constraint: Optional constraint for the beta weight.
+        gamma_constraint: Optional constraint for the gamma weight.
+        learned_epsilon: (bool) Whether to add another learnable
+        epsilon parameter or not.
+        name: Optional name for the layer
+
+    Input shape
```

Comment:
Can't it be extended to 5D?

Reply:
Theoretically it can be  but in paper they have proved it for 2D CNNs, so I think we can put 3D CNNs for hold as of now.

Reply:
Also, can you explain how this would work for fully connected layers? 

The output of an FC layer has shape `[B, N]` (where B is the batch shape, and N is the number of output neurons). Doing an operation (`tf.reduce_mean` in this case) over the axes `[1, 2]` will lead to an error.